src/CMakeLists.txt

```diff
@@ -18,10 +18,12 @@ else()
 endif()
 
 add_library(benchmark ${SOURCE_FILES} ${RE_FILES})
+find_package(Threads REQUIRED)
 
-# Link threading if building a shared library.
+# Link threading if building a shared library, otherwise let client do it.
 if (BUILD_SHARED_LIBS)
-  find_package(Threads REQUIRED)
+  target_link_libraries(benchmark PRIVATE ${CMAKE_THREAD_LIBS_INIT})
```

Comment:
If libbenchmark.so links to pthread.so then anything that links to benchmark must also link pthread in order to get the pthread symbols, otherwise it will fail to compile.

If libbenchmark.so links to libpthread.a then all of the symbols needed for benchmark are already provided and other targets do not need to link to pthread.

  -